Updates in gRPC and JDBC client

sangeetaraghu opened the pull request:
Hi @vietj ,

Updated the gRPC and JDBC client docs based on @pmlopes changes.
Also fixed the broken link.

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
+In {VertX} {v4}, the gRPC APIs support futures.  Using these APIs the gRPC plugin generates promisified applications. This makes it easier to reason about inputs and outputs.XXX
+
+The following example shows you XXX in {VertX} {v3x}.
+
```
@pmlopes Can you please help with text here?

pmlopes replied:
The following example shows the new `Future` api in use, in comparision to what was being generated in {Vertx} {v3x}.

sangeetaraghu replied:
As discussed @pmlopes used the following sentence:
The following example shows you use of promises in {VertX} {v3x}.

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
+
+== gRPC APIs support futures
+
+In {VertX} {v4}, the gRPC APIs support futures.  Using these APIs the gRPC plugin generates promisified applications. This makes it easier to reason about inputs and outputs.XXX
```
@pmlopes Can you please help me rephrase this line?

This makes it easier to reason about inputs and outputs.XXX

pmlopes replied:
The use of this API allows developers to clearly identify single asynchronous outputs of the service as a `Future` object. The same way, asynchronous streams are represented by a `WriteStream`. Optionally using this style, adapts the gRPC API to follow the same idioms used across all vert.x modules, making it easier to reason on a large vert.x application.

sangeetaraghu replied:
Thank you so much @pmlopes . As discussed using the following sentence:
In {VertX} {v4}, the gRPC APIs support futures. The gRPC plugin generates promisified APIs. These APIs use the standard {VertX} input and output arguments, which makes it easier to create standard {VertX} applications.

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
+----
+
+The following example shows you XXX in {VertX} {v4}.
+
```
@pmlopes Can you please help with text here?

pmlopes replied:
While in {VertX} {v3x} the greet response would be writen to an argument of the function, in {VertX} {v4} the asynchronous response is the method return value. Which reads simpler and the code flow is easy to follow.

sangeetaraghu replied:
As discussed @pmlopes used the following sentence:
The following example shows use of futures in {VertX} {v4}.

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:
```diff
@@ -1,8 +1,240 @@
+From {VertX} {v4}, the JDBC client supports SQL client. The SQL common module has also been merged in JDBC client, that is, `io.vertx:vertx-sql-common` merged in `io.vertx:vertx-jdbc-client` module. You will have to remove the `io.vertx:vertx-sql-common` dependency file because `io.vertx:vertx-jdbc-client` will include it. With the merging of SQL common client, all the database APIs have been con ...
+
+In {VertX} {v4}, the SQL client has been updated to include the following clients:XXX
+
```
@pmlopes Does this read correct?

In {VertX} {v4}, the SQL client has been updated to include the following clients:

pmlopes replied:
yes

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:
```diff
@@ -1,8 +1,240 @@
+
+The reactive implementations use the database network protocols. This makes them resource-efficient.
+
+JDBC calls to database are blocking calls. The JDBC client uses worker threads to make these calls non-blocking. XXX
```
@pmlopes Does this read correct?
JDBC calls to database are blocking calls. The JDBC client uses worker threads to make these calls non-blocking.

pmlopes replied:
:+1:

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:
```diff
@@ -1,8 +1,240 @@
+  });
+----
+
+== Support for Typsesafe Config XXX
```
@pmlopes Does this section read correct?

pmlopes replied:
:+1:

sangeetaraghu commented on asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:
```diff
@@ -1,8 +1,240 @@
+    );
+----
+
+NOTE: To use Typesafe Config, you need the `agroal` connection pool. The pool does not expose many configuration options and makes the configuration easy to use.XXX
```
@pmlopes Does this read correct?

pmlopes replied:
perhaps replace need with must:

```
NOTE: To use Typesafe Config, you must include the `agroal` connection pool in your project. The pool does not expose many configuration options and makes the configuration easy to use
```

sangeetaraghu replied:
Done

pmlopes commented on asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
 
 In previous releases, to work with gRPC, the following details were added to `pom.xml` file.
 
+[source,java]
```
it should be `[source,xml]`

pmlopes commented on asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
 
 In {VertX} {v4}, a new gRPC compiler plugin is available. This plugin uses the official gRPC compiler instead of the fork. To work with the new gRPC plugin, add the following details to `pom.xml` file.
 
+[source,java]
```
it should be `[source,xml]`

sangeetaraghu replied:
Done

sangeetaraghu commented:
- Updates in gRPC and JDBC client doc. 

- Added draft file for Opentracing

- Added clustering section:
https://github.com/vert-x3/vertx-4-migration-guide/issues/36

- Moved GraphQL under Common Components section:
https://github.com/vert-x3/vertx-4-migration-guide/issues/34
